From e14ab520ed4e012dbeb051eb9320e59f06873674 Mon Sep 17 00:00:00 2001
From: Mike Perry <mikeperry-git@fscked.org>
Date: Thu, 8 Sep 2011 08:40:17 -0700
Subject: [PATCH 08/12] Make content pref service memory-only + clearable

This prevents random urls from being inserted into content-prefs.sqllite in
the profile directory as content prefs change (includes site-zoom and perhaps
other site prefs?).
---
 .../contentprefs/nsContentPrefService.js           |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/toolkit/components/contentprefs/nsContentPrefService.js b/toolkit/components/contentprefs/nsContentPrefService.js
index 0d12b36..494515b 100644
--- a/toolkit/components/contentprefs/nsContentPrefService.js
+++ b/toolkit/components/contentprefs/nsContentPrefService.js
@@ -1183,7 +1183,7 @@ ContentPrefService.prototype = {
 
     var dbConnection;
 
-    if (!dbFile.exists())
+    if (true || !dbFile.exists())
       dbConnection = this._dbCreate(dbService, dbFile);
     else {
       try {
@@ -1231,7 +1231,7 @@ ContentPrefService.prototype = {
   },
 
   _dbCreate: function ContentPrefService__dbCreate(aDBService, aDBFile) {
-    var dbConnection = aDBService.openDatabase(aDBFile);
+    var dbConnection = aDBService.openSpecialDatabase("memory");
 
     try {
       this._dbCreateSchema(dbConnection);
-- 
1.7.3.4

